components/script/canvas_context.rs

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! Common interfaces for Canvas Contexts
+
+use canvas_traits::canvas::CanvasId;
+use euclid::default::Size2D;
+use ipc_channel::ipc::IpcSharedMemory;
+use script_layout_interface::{HTMLCanvasData, HTMLCanvasDataSource};
+
+use crate::dom::bindings::codegen::UnionTypes::HTMLCanvasElementOrOffscreenCanvas;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::node::{Node, NodeDamage};
+
+pub(crate) trait LayoutCanvasRenderingContextHelpers {
+    fn canvas_data_source(self) -> HTMLCanvasDataSource;
+}
+
+pub(crate) trait LayoutHTMLCanvasElementHelpers {
+    fn data(self) -> HTMLCanvasData;
+    fn get_canvas_id_for_layout(self) -> CanvasId;
```

Comment:
Rust typically does not use "get" in method names.
```suggestion
    fn canvas_id_for_layout(self) -> CanvasId;
```

Reply:
This trait is just copied from HTMLCanvasElement (to keep all context traits in same file), so I would not like to change that as part of this PR.